### PR TITLE
[dotnet] fix parsing of CustomResources

### PR DESCRIPTION
**Pull request by polivbr**
### Proposed changes

There is a bug in the .NET SDK where it attempts to create CustomResourceDefinitions rather than CustomResources
for unknown resource types. This PR fixes this.

### Related issues (optional)

https://github.com/pulumi/pulumi-kubernetes/issues/1120

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,15 +1,19 @@
 ## HEAD (Unreleased)
-(None)
+
+- [sdk/dotnet] Fix creation of CustomResources (https://github.com/pulumi/pulumi-kubernetes/pull/1741)
 
 ## 3.7.2 (September 17, 2021)
+
```
Comment (lblackstone) [inline CHANGELOG.md:6]: Let's revert the formatting changes to the changelog for this PR.

**Review by lblackstone — CHANGES_REQUESTED** [review]
Thanks for the PR, @polivbr!

@mikhailshilkov, could you review this? LGTM, but C# is not my strong suit.

File: CHANGELOG.md (hunk 1 above)
Comment (polivbr) [reply CHANGELOG.md:11]: Not sure where all of those formatting changes came from. I certainly didn't make them. That said, I'll patch it up.

Comment (polivbr) [reply CHANGELOG.md]: Ok. Changes reverted.

**Review by lblackstone — APPROVED** [review]

**Review by mikhailshilkov — APPROVED** [review]

**Comment by gitfool** [comment]
Interesting. Will this also fix https://github.com/pulumi/pulumi-kubernetes/issues/1225?

**Comment by lblackstone** [comment]
> Interesting. Will this also fix #1225?

Possibly -- I'm pretty surprised that this wasn't caught before, because it seems like it should have prevented a lot of Helm and YAML resources from working correctly in C#.

**Pull request merged**